<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yuxin Zou" />

<meta name="date" content="2019-10-07" />

<title>Fine-mapping Height ZBTB38 – SuSiE check</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">finemap-uk-biobank</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Fine-mapping Height ZBTB38 – SuSiE check</h1>
<h4 class="author"><em>Yuxin Zou</em></h4>
<h4 class="date"><em>10/07/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-11-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>finemap-uk-biobank/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191114code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191114)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191114code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191114)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfinemapukbiobanktreef71bb43c1cd3e6ac3762bcb267ff398c0c77d32dtargetblankf71bb43a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/finemap-uk-biobank/tree/f71bb43c1cd3e6ac3762bcb267ff398c0c77d32d" target="_blank">f71bb43</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfinemapukbiobanktreef71bb43c1cd3e6ac3762bcb267ff398c0c77d32dtargetblankf71bb43a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/height.ACAN.XtX.Xty.rds
    Untracked:  data/height.ADAMTS10.XtX.Xty.rds
    Untracked:  data/height.ADAMTS17.XtX.Xty.rds
    Untracked:  data/height.ADAMTSL3.XtX.Xty.rds
    Untracked:  data/height.CABLES1.XtX.Xty.rds
    Untracked:  data/height.CDK6.XtX.Xty.rds
    Untracked:  data/height.DLEU1.XtX.Xty.rds
    Untracked:  data/height.EFEMP1.XtX.Xty.rds
    Untracked:  data/height.GDF5.XtX.Xty.rds
    Untracked:  data/height.GNA12.XtX.Xty.rds
    Untracked:  data/height.HHIP.XtX.Xty.rds
    Untracked:  data/height.HMGA2.XtX.Xty.rds
    Untracked:  data/height.KDM2A.XtX.Xty.rds
    Untracked:  data/height.LCORL.XtX.Xty.rds
    Untracked:  data/height.MTMR11.XtX.Xty.rds
    Untracked:  data/height.UQCC1.XtX.Xty.rds
    Untracked:  data/height.ZBTB38.0.01.simulation1000.rds
    Untracked:  data/height.ZBTB38.XtX.Xty.rds
    Untracked:  data/height.ZBTB38.neale.rds
    Untracked:  data/height.ZBTB38.removeCS1.XtX.Xty.rds
    Untracked:  data/height.ZBTB38.removeCS2.XtX.Xty.rds
    Untracked:  data/height.ZBTB38.susie.model.rds
    Untracked:  output/height.CABLES1.removeCS1.XtX.Xty.rds
    Untracked:  output/height.CABLES1.removeCS2.XtX.Xty.rds
    Untracked:  output/height.GDF5.removeCS1.XtX.Xty.rds
    Untracked:  output/height.ZBTB38.plink2.height.glm.linear
    Untracked:  output/height.ZBTB38.removeCS1.XtX.Xty.rds
    Untracked:  output/height.ZBTB38.removeCS2.XtX.Xty.rds
    Untracked:  output/height.ZBTB38.simulation1000.rds
    Untracked:  output/height.ZBTB38.susie.model.rds

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/finemap_height_CABLES1_check.Rmd
    Modified:   analysis/index.Rmd
    Modified:   scripts/plots.R
    Modified:   scripts/prepare.region.sh
    Modified:   scripts/prepare.susieinput.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/f71bb43c1cd3e6ac3762bcb267ff398c0c77d32d/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">f71bb43</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-20
</td>
<td>
wflow_publish(“analysis/finemap_height_ZBTB38_check.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap-uk-biobank/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/finemap_height_ZBTB38_check.html" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/c14d255e6195d41367350eb93921f7198e64899b/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">c14d255</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
<td>
wflow_publish(“analysis/finemap_height_ZBTB38_check.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap-uk-biobank/9b47ac37e5262f739176d9e91278af8601e7f6ff/docs/finemap_height_ZBTB38_check.html" target="_blank">9b47ac3</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-17
</td>
<td>
move html files to docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/36beabbd91e5a496d1b8e38ed61dd5748b7c6263/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">36beabb</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-13
</td>
<td>
update result with inputation score filtered data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/4173d7ae66e18cd9e2f0a6e4a8650acc3f29605e/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">4173d7a</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-10-12
</td>
<td>
update plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/d6870cf758af664d9b8adb1eb57391d40ba5955b/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">d6870cf</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-10-12
</td>
<td>
update plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/89a3e6cd8faa961786a703746ae174c4f46c94ff/analysis/finemap_height_ZBTB38_check.Rmd" target="_blank">89a3e6c</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-10-09
</td>
<td>
change title; add detailed checks for 2 regions
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We perform some check for the SuSiE result on region around ZBTB38.</p>
<p>Load pacakges:</p>
<pre class="r"><code>library(readr)
library(dplyr)
library(gridExtra)
library(susieR)</code></pre>
<p>Load plotting functions:</p>
<pre class="r"><code>knitr::read_chunk(&quot;scripts/plots.R&quot;)</code></pre>
<pre class="r"><code>library(ggplot2)

#&#39; plot gene name annotations
#&#39; @param dat a matrix of gene names with &#39;start&#39; and &#39;end&#39; base-pair position
#&#39; @param xrange range of x axis, base-pair position
plot_geneName = function(dat, xrange, chr){
  ngene = 2:nrow(dat)
  line = 1
  dat$lines = NA
  dat$lines[1] = 1
  gene.end = dat[1, &#39;end&#39;]
  while(length(ngene) != 0){
    id = which(dat[ngene, &#39;start&#39;] &gt; gene.end + 0.02)[1]
    if(!is.na(id)){
      dat$lines[ngene[id]] = line
      gene.end = dat[ngene[id],&#39;end&#39;]
      ngene = ngene[-id]
    }else{
      line = line + 1
      dat$lines[ngene[1]] = line
      gene.end = dat[ngene[1],&#39;end&#39;]
      ngene = ngene[-1]
    }
  }
  
  dat$start = pmax(dat$start, xrange[1])
  dat$end = pmin(dat$end, xrange[2])
  dat$mean = rowMeans(dat[,c(&#39;start&#39;, &#39;end&#39;)])
  
  pl = ggplot(dat, aes(xmin = xrange[1], xmax = xrange[2])) + xlim(xrange[1], xrange[2]) + ylim(min(-dat$lines-0.6), -0.8) + 
    geom_rect(aes(xmin = start, xmax = end, ymin = -lines-0.05, ymax = -lines+0.05), fill=&#39;blue&#39;) +
    geom_text(aes(x = mean, y=-lines-0.4, label=geneName), size=4) + 
                xlab(paste0(&#39;base-pair position (Mb) on chromosome &#39;, chr)) + ylab(&#39;Gene&#39;) + 
    theme_bw() + theme(axis.text.x=element_blank(),
                       axis.ticks = element_blank(),
                       axis.text.y=element_blank(),
                       axis.title = element_text(size=15),
                       plot.title=element_text(size=11),
                       panel.grid.major = element_blank(),
                       panel.grid.minor = element_blank())
  pl
}

discrete_gradient_pal &lt;- function(colours, bins = 5) {
  ramp &lt;- scales::colour_ramp(colours)
  
  function(x) {
    if (length(x) == 0) return(character())
    
    i &lt;- floor(x * bins)
    i &lt;- ifelse(i &gt; bins-1, bins-1, i)
    ramp(i/(bins-1))
  }
}

scale_colour_discrete_gradient &lt;- function(..., colours, bins = 5, na.value = &quot;grey50&quot;, guide = &quot;colourbar&quot;, aesthetics = &quot;colour&quot;, colors)  {
  colours &lt;- if (missing(colours)) 
    colors
  else colours
  continuous_scale(
    aesthetics,
    &quot;discrete_gradient&quot;,
    discrete_gradient_pal(colours, bins),
    na.value = na.value,
    guide = guide,
    ...
  )
}

#&#39; Locuszoom plot
#&#39; @param z a vector of z scores with SNP names
#&#39; @param pos base-pair positions
#&#39; @param gene.pos.map a matrix of gene names with &#39;start&#39; and &#39;end&#39; base-pair position
#&#39; @param z.ref.name the reference SNP
#&#39; @param ld correlations between teh reference SNP and the rests
#&#39; @param title title of the plot
#&#39; @param title.size the size of the title
#&#39; @param true the true value
#&#39; @param y.height height of -log10(p) plot and height of the gene name annotation plot
#&#39; @param y.lim range of y axis
locus.zoom = function(z, pos, chr, gene.pos.map=NULL, z.ref.name=NULL, ld=NULL, 
                      title = NULL, title.size = 10, true = NULL, 
                      y.height=c(5,1.5), y.lim=NULL, y.type=&#39;logp&#39;,xrange=NULL){
  if(is.null(xrange)){
    xrange = c(min(pos), max(pos))
  }
  tmp = data.frame(POS = pos, p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10), z = z)
  
  if(!is.null(ld) &amp;&amp; !is.null(z.ref.name)){
    tmp$ref = names(z) == z.ref.name
    tmp$r2 = ld^2
    if(y.type == &#39;logp&#39;){
      pl_zoom = ggplot(tmp, aes(x = POS, y = p, shape = ref, size=ref, color=r2)) + geom_point() + 
        ylab(&quot;-log10(p value)&quot;) + ggtitle(title) + xlim(xrange[1], xrange[2]) +
        scale_color_gradientn(colors = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
                              values = seq(0,1,0.2), breaks=seq(0,1,0.2)) +
        # scale_colour_discrete_gradient(
        #   colours = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
        #   limits = c(0, 1.01),
        #   breaks = c(0,0.2,0.4,0.6,0.8,1),
        #   guide = guide_colourbar(nbin = 100, raster = FALSE, frame.colour = &quot;black&quot;, ticks.colour = NA)
        # ) + 
        scale_shape_manual(values=c(20, 18), guide=FALSE) + scale_size_manual(values=c(2,5), guide=FALSE) + 
        theme_bw() + theme(axis.title.x=element_blank(),
                           axis.title.y = element_text(size=15),axis.text = element_text(size=15),
                           plot.title = element_text(size=title.size))
    }else if(y.type == &#39;z&#39;){
      pl_zoom = ggplot(tmp, aes(x = POS, y = z, shape = ref, size=ref, color=r2)) + geom_point() + 
        ylab(&quot;z score&quot;) + ggtitle(title) + xlim(xrange[1], xrange[2]) +
        scale_color_gradientn(colors = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
                              values = seq(0,1,0.2), breaks=seq(0,1,0.2)) +
        # scale_colour_discrete_gradient(
        #   colours = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
        #   limits = c(0, 1.01),
        #   breaks = c(0,0.2,0.4,0.6,0.8,1),
        #   guide = guide_colourbar(nbin = 100, raster = FALSE, frame.colour = &quot;black&quot;, ticks.colour = NA)
        # ) + 
        scale_shape_manual(values=c(20, 18), guide=FALSE) + scale_size_manual(values=c(2,5), guide=FALSE) + 
        theme_bw() + theme(axis.title.x=element_blank(),
                           axis.title.y = element_text(size=15),axis.text = element_text(size=15),
                           plot.title = element_text(size=title.size))
    }
  }else{
    if(y.type == &#39;logp&#39;){
      pl_zoom = ggplot(tmp, aes(x = POS, y = p)) + geom_point(color = &#39;darkblue&#39;) + 
        ylab(&quot;-log10(p value)&quot;) + ggtitle(title) + xlim(xrange[1], xrange[2]) +
        theme_bw() + theme(axis.title.x=element_blank(),
                           plot.title = element_text(size=title.size))
    }else if(y.type == &#39;z&#39;){
      pl_zoom = ggplot(tmp, aes(x = POS, y = z)) + geom_point(color = &#39;darkblue&#39;) + 
        ylab(&quot;z scores&quot;) + ggtitle(title) + xlim(xrange[1], xrange[2]) +
        theme_bw() + theme(axis.title.x=element_blank(),
                           plot.title = element_text(size=title.size))
    }
  }
  
  if(!is.null(y.lim)){
    pl_zoom = pl_zoom + ylim(y.lim[1], y.lim[2])
  }
  # pl_zoom = pl_zoom + geom_hline(yintercept=-log10(5e-08), linetype=&#39;dashed&#39;, color = &#39;red&#39;)
  if(!is.null(true)){
    tmp.true = data.frame(POS = which(true!=0), p = tmp$p[which(true!=0)], 
                          ref = (names(z) == z.ref.name)[which(true!=0)],
                          label = paste0(&#39;SNP&#39;,1:length(which(true!=0))))
    pl_zoom = pl_zoom + geom_point(data=tmp.true, aes(x=POS, y=p), 
                                   color=&#39;red&#39;, show.legend = FALSE, shape=1, stroke = 1) + 
      geom_text(data=tmp.true, aes(x = POS-30, y=p+1, label=label), size=3, color=&#39;red&#39;)
  }
  if(!is.null(gene.pos.map)){
    pl_gene = plot_geneName(gene.pos.map, xrange = xrange, chr=chr)
    g = egg::ggarrange(pl_zoom, pl_gene, nrow=2, heights = y.height, draw=FALSE)
  }else{
    g = pl_zoom
  }
  g
}

#&#39; SuSiE plot with Locuszoom plot
#&#39; @param z a vector of z scores with SNP names
#&#39; @param model the fitted SuSiE model
#&#39; @param pos base-pair positions
#&#39; @param gene.pos.map a matrix of gene names with &#39;start&#39; and &#39;end&#39; base-pair position
#&#39; @param z.ref.name the reference SNP
#&#39; @param ld correlations between teh reference SNP and the rests
#&#39; @param title title of the plot
#&#39; @param title.size the size of the title
#&#39; @param true the true value
#&#39; @param plot.locuszoom whether to plot locuszoom plot
#&#39; @param y.lim range of y axis
#&#39; @param y.susie the y axis of the SuSiE plot, &#39;PIP&#39; or &#39;p&#39; or &#39;z&#39;, &#39;p&#39; refers to -log10(p)
susie_plot_locuszoom = function(z, model, pos, chr, gene.pos.map = NULL, z.ref.name, ld, 
                                title = NULL, title.size = 10, true = NULL, 
                                plot.locuszoom = TRUE, y.lim=NULL, y.susie=&#39;PIP&#39;, xrange=NULL){
  if(is.null(xrange)){
    xrange = c(min(pos), max(pos))
  }
  if(plot.locuszoom){
    if(y.susie == &#39;z&#39;){
      y.type = &#39;z&#39;
    }else{
      y.type = &#39;logp&#39;
    }
    pl_zoom = locus.zoom(z, pos = pos, chr = chr, ld=ld, z.ref.name = z.ref.name, title = title, title.size = title.size, y.lim=y.lim, y.type=y.type, xrange=xrange)
  }
  pip = model$pip
  tmp = data.frame(POS = pos, PIP = pip, p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10), z = z)
  if(y.susie == &#39;PIP&#39;){
    pl_susie = ggplot(tmp, aes(x = POS, y = PIP)) + geom_point(show.legend = FALSE, size=3) + 
      xlim(xrange[1], xrange[2]) + 
      theme_bw() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(),axis.text = element_text(size=15),
                         axis.title.y = element_text(size=15))
    if(!plot.locuszoom){
      pl_susie = pl_susie + ggtitle(title) + theme(plot.title = element_text(size=title.size))
    }
  }else if(y.susie == &#39;p&#39;){
    pl_susie = ggplot(tmp, aes(x = POS, y = p)) + geom_point(show.legend = FALSE, size=3) + 
      ylab(&quot;-log10(p value)&quot;) + xlim(xrange[1], xrange[2]) + 
      theme_bw() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(),axis.text = element_text(size=15),
                         axis.title.y = element_text(size=15))
    if(!plot.locuszoom){
      pl_susie = pl_susie + ggtitle(title) + theme(plot.title = element_text(size=title.size))
      # pl_susie = pl_susie + geom_hline(yintercept=-log10(5e-08), linetype=&#39;dashed&#39;, color = &#39;red&#39;)
    }
  }else if(y.susie == &#39;z&#39;){
    pl_susie = ggplot(tmp, aes(x = POS, y = z)) + geom_point(show.legend = FALSE, size=3) + 
      ylab(&quot;z scores&quot;) + xlim(xrange[1], xrange[2]) + 
      theme_bw() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(),axis.text = element_text(size=15),
                         axis.title.y = element_text(size=15))
    if(!plot.locuszoom){
      pl_susie = pl_susie + ggtitle(title) + theme(plot.title = element_text(size=title.size))
    }
  }
  
  if(!is.null(true)){
    tmp.true = data.frame(POS = pos[which(true!=0)], PIP = pip[which(true!=0)])
    pl_susie = pl_susie + geom_point(data=tmp.true, aes(x=POS, y=PIP), 
                                     color=&#39;red&#39;, size=3, show.legend = FALSE)
  }
  
  model.cs = model$sets$cs
  if(!is.null(model.cs)){
    tmp$CS = numeric(length(z))
    for(i in 1:length(model.cs)){
      tmp$CS[model.cs[[i]]] = gsub(&#39;L&#39;, &#39;CS&#39;, names(model.cs)[i])
    }
    tmp.cs = tmp[unlist(model.cs),]
    tmp.cs$CS = factor(tmp.cs$CS)
    levels(tmp.cs$CS) = paste0(&#39;CS&#39;, 1:length(model.cs))
    colors = c(&#39;red&#39;, &#39;cyan&#39;, &#39;green&#39;, &#39;orange&#39;, &#39;dodgerblue&#39;, &#39;violet&#39;, &#39;gold&#39;,
               &#39;#FF00FF&#39;, &#39;forestgreen&#39;, &#39;#7A68A6&#39;)
    if(y.susie == &#39;PIP&#39;){
      pl_susie = pl_susie + geom_point(data=tmp.cs, aes(x=POS, y=PIP, color=CS), 
                                       size=3, shape=1, stroke = 2) + 
        scale_color_manual(values=colors)
    }else if(y.susie == &#39;p&#39;){
      pl_susie = pl_susie + geom_point(data=tmp.cs, aes(x=POS, y=p, color=CS), 
                                       shape=1, size=3, stroke=1.5) + 
        scale_color_manual(values=colors)
    }else if(y.susie == &#39;z&#39;){
      pl_susie = pl_susie + geom_point(data=tmp.cs, aes(x=POS, y=z, color=CS), 
                                       shape=1, size=3, stroke=1.5) + 
        scale_color_manual(values=colors)
    }
  }
  
  if(!is.null(gene.pos.map)){
    pl_gene = plot_geneName(gene.pos.map, xrange = xrange, chr=chr)
    if(plot.locuszoom){
      g = egg::ggarrange(pl_zoom, pl_susie, pl_gene, nrow=3, heights = c(4,4,1.5), draw=FALSE)
    }else{
      g = egg::ggarrange(pl_susie, pl_gene, nrow=2, heights = c(5.5,1.5), draw=FALSE)
    }
    
  }else{
    if(plot.locuszoom){
      g = egg::ggarrange(pl_zoom, pl_susie, nrow=2, heights = c(4,4), draw=FALSE)
    }else{
      g = pl_susie
    }
  }
  g
}</code></pre>
<pre class="r"><code>locus.zoom.cs = function(z, cs, pos, chr, gene.pos.map=NULL, z.ref.name=NULL, ld=NULL, title = NULL, title.size = 10, xrange = NULL, y.lab=&#39;-log10(p value)&#39;, y.type = &#39;logp&#39;){
  if(is.null(xrange)){
    xrange = c(min(pos), max(pos))
  }
  tmp = data.frame(POS = pos, log10p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10), z = z)
  tmp$ref = names(z) == z.ref.name
  tmp$r2 = ld^2
  tmp$CS = rep(4, length(z))
  tmp$CS[cs] = 16
  tmp$CS = as.factor(tmp$CS)
  if(y.type == &#39;logp&#39;){
    pl_zoom = ggplot(tmp, aes(x = POS, y = log10p, shape = CS, color = r2, size=CS)) + 
    geom_point() + ylab(y.lab)
  }else if(y.type == &#39;z&#39;){
    pl_zoom = ggplot(tmp, aes(x = POS, y = z, shape = CS, color = r2, size=CS)) + 
    geom_point() + ylab(y.lab)
  }
  pl_zoom = pl_zoom + scale_color_gradientn(colors = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
                          values = seq(0,1,0.2), breaks=seq(0,1,0.2)) + 
    # scale_colour_discrete_gradient(
    #     colours = c(&quot;darkblue&quot;, &quot;deepskyblue&quot;, &quot;lightgreen&quot;, &quot;orange&quot;, &quot;red&quot;),
    #     limits = c(0, 1.01),
    #     breaks = c(0,0.2,0.4,0.6,0.8,1),
    #     guide = guide_colourbar(nbin = 100, raster = FALSE, frame.colour = &quot;black&quot;, ticks.colour = NA)) +
    scale_shape_manual(values = c(4, 19), guide=FALSE) + 
    scale_size_manual(values=c(1.5,4), guide=FALSE) + 
    ggtitle(title) + 
    theme_bw() + theme(axis.title.x=element_blank(), axis.text=element_text(size=15),
                       axis.title.y=element_text(size=12),
                       plot.title = element_text(size=title.size))
  tmp.sub = tmp[cs,]
  if(y.type == &#39;logp&#39;){
    pl_zoom = pl_zoom + geom_point(data = tmp.sub, aes(x=POS, y=log10p), shape=1, size=4, color=&#39;black&#39;, stroke=0.1)
  }else if(y.type == &#39;z&#39;){
    pl_zoom = pl_zoom + geom_point(data = tmp.sub, aes(x=POS, y=z), shape=1, size=4, color=&#39;black&#39;, stroke=0.1)
  }
  
  if(!is.null(xrange)){
    pl_zoom = pl_zoom + xlim(xrange[1], xrange[2])
  }
  
  pl_gene = plot_geneName(gene.pos.map, xrange = xrange, chr=chr)
  g = egg::ggarrange(pl_zoom, pl_gene, nrow=2, heights = c(5.5,1.5), draw=FALSE)
  g
}</code></pre>
<p>Get summary statistics:</p>
<pre class="r"><code>ss.dat = readRDS(&#39;data/height.ZBTB38.XtX.Xty.rds&#39;)
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * as.vector(ss.dat$Xty)
se = sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX)))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)</code></pre>
<p>Get gene data:</p>
<pre class="r"><code>genes        &lt;- read_delim(&quot;data/seq_gene.md.gz&quot;,delim = &quot;\t&quot;,quote = &quot;&quot;)
class(genes) &lt;- &quot;data.frame&quot;
genes        &lt;- subset(genes,
                       group_label == &quot;GRCh37.p5-Primary Assembly&quot; &amp;
                       feature_type == &quot;GENE&quot;)
start.pos &lt;- min(ss.dat$pos$POS)
stop.pos  &lt;- max(ss.dat$pos$POS)
plot.genes &lt;- subset(genes,
                     chromosome == 3 &amp;
                     ((chr_start &gt; start.pos &amp; chr_start &lt; stop.pos) |
                      (chr_stop &gt; start.pos &amp; chr_start &lt; stop.pos)) &amp; feature_type == &#39;GENE&#39;)
gene.pos.map = plot.genes %&gt;% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c(&#39;geneName&#39;, &#39;start&#39;, &#39;end&#39;)
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %&gt;% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(5,10,11,16),]</code></pre>
<p>SuSiE bhat with standardize result: the top panel shows the r2 with respect to the top hit (diamond shape); the lower panel plots the credible sets in PIP, -log10(p value) and z scores.</p>
<pre class="r"><code>mod_bhat = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)), track_fit=TRUE, standardize = T)
z.max = which.max(abs(z))
p1 = susie_plot_locuszoom(z, mod_bhat, pos = ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = &#39;rs2871960_C&#39;)
p2 = susie_plot_locuszoom(z, mod_bhat, pos = ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = &#39;rs2871960_C&#39;, y.susie =&#39;p&#39;)
grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_standardize_results-1.png" width="2880" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ZBTB38_bhat_standardize_results-1">
Past versions of ZBTB38_bhat_standardize_results-1.png
</button>
</p>
<div id="fig-ZBTB38_bhat_standardize_results-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_standardize_results-1.png" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For 1Mb region about ZBTB38, SuSiE found 2 CSs. The SNP with the strongest marginal p value in CS1 is rs2871960 (p = 1.8670e-207). For CS2, the SNP with the strongest marginal p value is rs11919556 (p = 0.0211).</p>
<p>The correlation between rs2871960 and rs11919556 is 0.221051. The average correlation between SNPs in CS1 and CS2 is</p>
<pre class="r"><code>round(mean(abs(R[mod_bhat$sets$cs$L1, mod_bhat$sets$cs$L2])), 4)</code></pre>
<pre><code>[1] 0.1812</code></pre>
<p>Zoom in plot:</p>
<pre class="r"><code>p1 = susie_plot_locuszoom(z, mod_bhat, pos = ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, z.ref.name = names(z), ld = R[z.max,], title=&#39;ZBTB38 Credible Sets&#39;, plot.locuszoom = FALSE, y.susie = &#39;p&#39;, xrange=c(140.9,141.5), title.size = 20)
p1</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_standardize_CSs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ZBTB38_bhat_standardize_CSs-1">
Past versions of ZBTB38_bhat_standardize_CSs-1.png
</button>
</p>
<div id="fig-ZBTB38_bhat_standardize_CSs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_standardize_CSs-1.png" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The first credible set contains</p>
<pre class="r"><code>cs1 = ss.dat$pos[unlist(mod_bhat$sets$cs$L1),]
cs1$gene = sapply(cs1$POS/1e6, function(i){
  id = intersect(which(gene.pos.map$start &lt;= i), which(gene.pos.map$end &gt;= i))
  gene.pos.map$geneName[id]
})
cs1</code></pre>
<pre><code>     #CHROM       POS         ID REF ALT      maf   gene
1255      3 141101961  rs9853018   T   C 0.446589 ZBTB38
1258      3 141102833  rs6763931   A   G 0.446887 ZBTB38
1261      3 141105570   rs724016   G   A 0.447055 ZBTB38
1262      3 141106063  rs7632381   C   T 0.447588 ZBTB38
1267      3 141109321  rs6808936   G   A 0.447482 ZBTB38
1268      3 141109348  rs6785012   T   C 0.447476 ZBTB38
1270      3 141110074  rs4683606   G   A 0.447216 ZBTB38
1278      3 141118028 rs13068733   G   A 0.447268 ZBTB38
1281      3 141121814  rs2871960   C   A 0.447921 ZBTB38
1288      3 141125186  rs1344674   G   A 0.449007 ZBTB38
1291      3 141125705  rs1344672   G   C 0.448705 ZBTB38</code></pre>
<p>The second credible set contains</p>
<pre class="r"><code>cs2 = ss.dat$pos[unlist(mod_bhat$sets$cs$L2),]
cs2$gene = sapply(cs2$POS/1e6, function(i){
  id = intersect(which(gene.pos.map$start &lt;= i), which(gene.pos.map$end &gt;= i))
  gene.pos.map$geneName[id]
})
cs2</code></pre>
<pre><code>     #CHROM       POS          ID REF ALT      maf   gene
1225      3 141089418 rs115198977   A   C 0.028305 ZBTB38
1226      3 141091356 rs116086535   T   G 0.039387 ZBTB38
1230      3 141092497  rs55675250   C   T 0.028124 ZBTB38
1231      3 141092645  rs56259972   T   A 0.028307 ZBTB38
1234      3 141094334  rs77022697   T   C 0.027163 ZBTB38
1253      3 141101781  rs80060248   C   G 0.027312 ZBTB38
1256      3 141102130 rs118108151   A   G 0.027106 ZBTB38
1269      3 141110025  rs79361824   G   A 0.027174 ZBTB38
1275      3 141114894  rs75828411   A   G 0.027178 ZBTB38
1280      3 141121557 rs117589471   C   G 0.027313 ZBTB38
1284      3 141123883 rs117323650   T   C 0.027176 ZBTB38
1301      3 141131707  rs75092195   A   G 0.027288 ZBTB38
1303      3 141133260  rs74441190   A   G 0.027288 ZBTB38
1312      3 141135004  rs11919556   C   T 0.041658 ZBTB38
1315      3 141135690 rs114626934   G   A 0.027292 ZBTB38
1317      3 141136022   rs6440005   C   T 0.027713 ZBTB38
1324      3 141138545 rs115126368   C   T 0.027296 ZBTB38
1325      3 141138692 rs144648746   A   G 0.027296 ZBTB38
1344      3 141143205  rs79057647   C   T 0.027288 ZBTB38
1347      3 141143489  rs77828701   C   T 0.025980 ZBTB38
1348      3 141143491  rs76300426   T   C 0.026070 ZBTB38
1350      3 141143620  rs79197573   G   A 0.027285 ZBTB38
1360      3 141145479  rs79311431   G   A 0.027700 ZBTB38
1361      3 141145601  rs79153193   A   C 0.027680 ZBTB38
1366      3 141147784 rs115605995   A   C 0.027669 ZBTB38
1390      3 141155631 rs140143440   T   C 0.027114 ZBTB38
1391      3 141156007 rs114802788   T   C 0.027317 ZBTB38
1474      3 141197995 rs116215663   C   A 0.025722       </code></pre>
<div id="cs-1" class="section level2">
<h2>CS 1</h2>
<ul>
<li>Option 1. We remove effect of top SNP in CS 2.</li>
</ul>
<pre class="r"><code>library(data.table)
library(readr)
library(Matrix)
geno.file = &#39;height.ZBTB38.raw.gz&#39;
cat(&quot;Reading genotype data.\n&quot;)
geno &lt;- fread(geno.file,sep = &quot;\t&quot;,header = TRUE,stringsAsFactors = FALSE)
class(geno) &lt;- &quot;data.frame&quot;

# Extract the genotypes.
X &lt;- as(as.matrix(geno[-(1:6)]),&#39;dgCMatrix&#39;)

pheno.file &lt;- &quot;/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz&quot;
out.pheno.file &lt;- &quot;pheno.height.txt&quot;
out.covar.file &lt;- &quot;covar.removeCS2.height.txt&quot;

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat(&quot;Reading phenotype data.\n&quot;)
pheno        &lt;- suppressMessages(read_csv(pheno.file))
class(pheno) &lt;- &quot;data.frame&quot;
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread(&#39;height.ZBTB38.psam&#39;)
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20 + X[,1312], data = pheno)

# Remove intercept
Z = Z[,-1]
colnames(Z)[150] = &#39;X&#39;
Z = scale(Z, center=T, scale=F)
# standardize quantitative columns
cols = which(colnames(Z) %in% c(&quot;age&quot;,&quot;pc_genetic1&quot;,&quot;pc_genetic2&quot;,&quot;pc_genetic3&quot;,&quot;pc_genetic4&quot;,
                                &quot;pc_genetic5&quot;,&quot;pc_genetic6&quot;,&quot;pc_genetic7&quot;,&quot;pc_genetic8&quot;,&quot;pc_genetic9&quot;, 
                                &quot;pc_genetic10&quot;,&quot;pc_genetic11&quot;,&quot;pc_genetic12&quot;,&quot;pc_genetic13&quot;,&quot;pc_genetic14&quot;,
                                &quot;pc_genetic15&quot;,&quot;pc_genetic16&quot;,&quot;pc_genetic17&quot;,&quot;pc_genetic18&quot;,&quot;pc_genetic19&quot;,&quot;pc_genetic20&quot;))
Z[,cols] = scale(Z[,cols])
Z[,&#39;age2&#39;] = Z[,&#39;age&#39;]^2

# Compute XtX and Xty
y = pheno$height
names(y) = pheno$id

# Center y
y = y - mean(y)
# Center X
X = scale(X, center=T, scale=FALSE)
xtxdiag = colSums(X^2)

A   &lt;- crossprod(Z) # Z&#39;Z
# chol decomposition for (Z&#39;Z)^(-1)
R = chol(solve(A)) # R&#39;R = (Z&#39;Z)^(-1)
W = R %*% crossprod(Z, X) # RZ&#39;X
S = R %*% crossprod(Z, y) # RZ&#39;y

# Load LD matrix from raw genotype
ld.matrix = as.matrix(fread(paste0(&#39;height.ZBTB38.matrix&#39;)))
# X&#39;X
XtX = sqrt(xtxdiag) * t(ld.matrix*sqrt(xtxdiag)) - crossprod(W) # W&#39;W = X&#39;ZR&#39;RZ&#39;X = X&#39;Z(Z&#39;Z)^{-1}Z&#39;X
rownames(XtX) = colnames(XtX) = colnames(X)
# X&#39;y
Xty = as.vector(y %*% X)
Xty = Xty - crossprod(W, S) # W&#39;S = X&#39;ZR&#39;RZ&#39;y = X&#39;Z(Z&#39;Z)^{-1}Z&#39;y

## SNP info
maf &lt;- read.delim(&#39;height.ZBTB38.afreq&#39;)
pos &lt;- fread(&#39;height.ZBTB38.pvar&#39;)
pos$maf = pmin(maf$ALT_FREQS, 1-maf$ALT_FREQS)

saveRDS(list(XtX = XtX, Xty = Xty, yty = sum(y^2) - crossprod(S), n = length(y), pos=pos),
        paste0(&#39;height.ZBTB38.removeCS2.XtX.Xty.rds&#39;))</code></pre>
<p>After removing the effect of rs11919556 (p = 0.0211), the p values and z scores are plotted below. The color is correspongding to LD, the shape is corresponding to CS. The SNP in CS is labeled with filled circle.</p>
<pre class="r"><code>ss.dat = readRDS(&#39;output/height.ZBTB38.removeCS2.XtX.Xty.rds&#39;)
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
z.max = which.max(abs(z))
p1 = locus.zoom.cs(z, cs = mod_bhat$sets$cs$L1, pos=ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, z.ref.name = names(z)[z.max], ld = R[z.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 CS1&#39;, y.lab = &#39;-log10(p value condition on CS2)&#39;)
p2 = locus.zoom.cs(z, cs = mod_bhat$sets$cs$L1, pos=ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, z.ref.name = names(z)[z.max], ld = R[z.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 CS1&#39;, y.lab = &#39;z scores condition on CS2&#39;, y.type = &#39;z&#39;)
grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_nostandardize_CS1_2-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ZBTB38_bhat_nostandardize_CS1_2-1">
Past versions of ZBTB38_bhat_nostandardize_CS1_2-1.png
</button>
</p>
<div id="fig-ZBTB38_bhat_nostandardize_CS1_2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_nostandardize_CS1_2-1.png" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Option 2. We remove effect of all other CSs.</li>
</ul>
<p>Instead of removing the effect of top SNP from other CSs, we do exactly what SuSiE does here. In SuSiE, we estimate effects using residuals that are obtained by removing the effects of all other CSs.</p>
<p>The residuals after removing the effects from CSs other than CS1 is <span class="math display">\[
r = y - \mathbf{X} \sum_{l=2}^{L}\hat{\mathbf{b}}_{l}.
\]</span></p>
<pre class="r"><code>ss.dat = readRDS(&#39;data/height.ZBTB38.XtX.Xty.rds&#39;)
XtX.scale = t(ss.dat$XtX / mod_bhat$X_column_scale_factors) / mod_bhat$X_column_scale_factors
XtXr = mod_bhat$XtXr - XtX.scale %*% (mod_bhat$alpha[1,] * mod_bhat$mu[1,]) 
Xtr = ss.dat$Xty/mod_bhat$X_column_scale_factors - XtXr
betas = Xtr/diag(XtX.scale)
b_2 = colSums(mod_bhat$alpha[-1,]*mod_bhat$mu[-1,])
rss = c(ss.dat$yty - 2*sum(ss.dat$Xty * b_2) + sum(XtXr * b_2)) - betas * Xtr
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(XtX.scale))))
z.CS1 = betas/se
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z.CS1) = colnames(R)
z.cs1.max = which.max(abs(z.CS1))
p1 = locus.zoom.cs(z.CS1, mod_bhat$sets$cs$L1, pos=ss.dat$pos$POS/1e6, chr = 3, gene.pos.map = gene.pos.map, z.ref.name = names(z.cs1.max), ld = R[z.cs1.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 Credible Set 1&#39;, y.lab = &#39;-log10(p value condition on other CSs)&#39;, title.size = 20)
p2 = locus.zoom.cs(z.CS1, mod_bhat$sets$cs$L1, pos=ss.dat$pos$POS/1e6, chr = 3, gene.pos.map = gene.pos.map, z.ref.name = names(z.cs1.max), ld = R[z.cs1.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 Credible Set 1&#39;, y.lab = &#39;z scores condition on other CSs&#39;, title.size = 20, y.type=&#39;z&#39;)
grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/unnamed-chunk-10-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/figure/finemap_height_ZBTB38_check.Rmd/unnamed-chunk-10-1.png" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cs-2" class="section level2">
<h2>CS 2</h2>
<ul>
<li>Option 1. We remove effect of top SNP in CS 1.</li>
</ul>
<pre class="r"><code>library(data.table)
library(readr)
library(Matrix)
geno.file = &#39;height.ZBTB38.raw.gz&#39;
cat(&quot;Reading genotype data.\n&quot;)
geno &lt;- fread(geno.file,sep = &quot;\t&quot;,header = TRUE,stringsAsFactors = FALSE)
class(geno) &lt;- &quot;data.frame&quot;

# Extract the genotypes.
X &lt;- as(as.matrix(geno[-(1:6)]),&#39;dgCMatrix&#39;)

pheno.file &lt;- &quot;/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz&quot;
out.pheno.file &lt;- &quot;pheno.height.txt&quot;
out.covar.file &lt;- &quot;covar.removeCS1.height.txt&quot;

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat(&quot;Reading phenotype data.\n&quot;)
pheno        &lt;- suppressMessages(read_csv(pheno.file))
class(pheno) &lt;- &quot;data.frame&quot;
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread(&#39;height.ZBTB38.psam&#39;)
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20 + X[,1281], data = pheno)

# Remove intercept
Z = Z[,-1]
colnames(Z)[150] = &#39;X&#39;
Z = scale(Z, center=T, scale = F)
# standardize quantitative columns
cols = which(colnames(Z) %in% c(&quot;age&quot;,&quot;pc_genetic1&quot;,&quot;pc_genetic2&quot;,&quot;pc_genetic3&quot;,&quot;pc_genetic4&quot;,
                                &quot;pc_genetic5&quot;,&quot;pc_genetic6&quot;,&quot;pc_genetic7&quot;,&quot;pc_genetic8&quot;,&quot;pc_genetic9&quot;, 
                                &quot;pc_genetic10&quot;,&quot;pc_genetic11&quot;,&quot;pc_genetic12&quot;,&quot;pc_genetic13&quot;,&quot;pc_genetic14&quot;,
                                &quot;pc_genetic15&quot;,&quot;pc_genetic16&quot;,&quot;pc_genetic17&quot;,&quot;pc_genetic18&quot;,&quot;pc_genetic19&quot;,&quot;pc_genetic20&quot;))
Z[,cols] = scale(Z[,cols])
Z[,&#39;age2&#39;] = Z[,&#39;age&#39;]^2

# Compute XtX and Xty
y = pheno$height
names(y) = pheno$id

# Center y
y = y - mean(y)
# Center X
X = scale(X, center=TRUE, scale = FALSE)
xtxdiag = colSums(X^2)

A   &lt;- crossprod(Z) # Z&#39;Z
# chol decomposition for (Z&#39;Z)^(-1)
R = chol(solve(A)) # R&#39;R = (Z&#39;Z)^(-1)
W = R %*% crossprod(Z, X) # RZ&#39;X
S = R %*% crossprod(Z, y) # RZ&#39;y

# Load LD matrix from raw genotype
ld.matrix = as.matrix(fread(paste0(&#39;height.ZBTB38.matrix&#39;)))
# X&#39;X
XtX = sqrt(xtxdiag) * t(ld.matrix*sqrt(xtxdiag)) - crossprod(W) # W&#39;W = X&#39;ZR&#39;RZ&#39;X = X&#39;Z(Z&#39;Z)^{-1}Z&#39;X
rownames(XtX) = colnames(XtX) = colnames(X)
# X&#39;y
Xty = as.vector(y %*% X)
Xty = Xty - crossprod(W, S) # W&#39;S = X&#39;ZR&#39;RZ&#39;y = X&#39;Z(Z&#39;Z)^{-1}Z&#39;y

## SNP info
maf &lt;- read.delim(&#39;height.ZBTB38.afreq&#39;)
pos &lt;- fread(&#39;height.ZBTB38.pvar&#39;)
pos$maf = pmin(maf$ALT_FREQS, 1-maf$ALT_FREQS)

saveRDS(list(XtX = XtX, Xty = Xty, yty = sum(y^2) - crossprod(S), n = length(y), pos=pos),
        paste0(&#39;height.ZBTB38.removeCS1.XtX.Xty.rds&#39;))</code></pre>
<p>After removing the effect of rs2871960 (p = 1.8674e-207), the conditional p value for rs11919556 becomes 4.279e-06, and it becomes the strongest one among all SNPs!</p>
<pre class="r"><code>ss.dat = readRDS(&#39;output/height.ZBTB38.removeCS1.XtX.Xty.rds&#39;)
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
z.max = which.max(abs(z))
p1 = locus.zoom.cs(z, cs = mod_bhat$sets$cs$L2, pos=ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, z.ref.name = names(z)[z.max], ld = R[z.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 CS2&#39;, y.lab = &#39;-log10(p value condition on CS1)&#39;)
p2 = locus.zoom.cs(z, cs = mod_bhat$sets$cs$L2, pos=ss.dat$pos$POS/1e6, chr=3, gene.pos.map = gene.pos.map, z.ref.name = names(z)[z.max], ld = R[z.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 CS2&#39;, y.lab = &#39;z scores condition on CS1&#39;, y.type = &#39;z&#39;)
grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_nostandardize_CS2_1-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ZBTB38_bhat_nostandardize_CS2_1-1">
Past versions of ZBTB38_bhat_nostandardize_CS2_1-1.png
</button>
</p>
<div id="fig-ZBTB38_bhat_nostandardize_CS2_1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap-uk-biobank/blob/149b7c77f68d982f87ba5d2e10fee342da10f9f6/docs/figure/finemap_height_ZBTB38_check.Rmd/ZBTB38_bhat_nostandardize_CS2_1-1.png" target="_blank">149b7c7</a>
</td>
<td>
zouyuxin
</td>
<td>
2019-11-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>Option 2. We remove effect of all other CSs.</li>
</ul>
<p>Instead of removing the effect of top SNP from other CSs, we do exactly what SuSiE does here. In SuSiE, we estimate effects using residuals that are obtained by removing the effects of all other CSs.</p>
<p>The residuals after removing the effects from CSs other than CS2 is <span class="math display">\[
r = y - \mathbf{X} \sum_{l\neq 2}^{L}\hat{\mathbf{b}}_{l}.
\]</span></p>
<pre class="r"><code>ss.dat = readRDS(&#39;data/height.ZBTB38.XtX.Xty.rds&#39;)
XtX.scale = t(ss.dat$XtX / mod_bhat$X_column_scale_factors) / mod_bhat$X_column_scale_factors
XtXr = mod_bhat$XtXr - XtX.scale %*% (mod_bhat$alpha[2,] * mod_bhat$mu[2,]) 
Xtr = ss.dat$Xty/mod_bhat$X_column_scale_factors - XtXr
betas = Xtr/diag(XtX.scale)
b_2 = colSums(mod_bhat$alpha[-2,]*mod_bhat$mu[-2,])
rss = c(ss.dat$yty - 2*sum(ss.dat$Xty * b_2) + sum(XtXr * b_2)) - betas * Xtr
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(XtX.scale))))
z.CS2 = betas/se
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z.CS2) = colnames(R)
z.cs2.max = which.max(abs(z.CS2))
p1 = locus.zoom.cs(z.CS2, mod_bhat$sets$cs$L2, pos=ss.dat$pos$POS/1e6, chr = 3, gene.pos.map = gene.pos.map, z.ref.name = names(z.cs2.max), ld = R[z.cs2.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 Credible Set 2&#39;, y.lab = &#39;-log10(p value condition on other CSs)&#39;, title.size = 20)
p2 = locus.zoom.cs(z.CS2, mod_bhat$sets$cs$L2, pos=ss.dat$pos$POS/1e6, chr = 3, gene.pos.map = gene.pos.map, z.ref.name = names(z.cs2.max), ld = R[z.cs2.max,], xrange=c(140.9,141.5), title=&#39;ZBTB38 Credible Set 2&#39;, y.lab = &#39;z scores condition on other CSs&#39;, title.size = 20, y.type=&#39;z&#39;)
grid.arrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/finemap_height_ZBTB38_check.Rmd/unnamed-chunk-12-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="simulation-under-the-estimated-model" class="section level2">
<h2>Simulation under the estimated model</h2>
<p>In the following simulation, we treat rs2871960 and rs11919556 as true signals. The effect sizes are from estimated model. We use the fitted residual variance in the simulation. The response y is simulated from <span class="math display">\[
y \sim N_n(Xb, \sigma^2 I)
\]</span> , where X the genotype matrix that column centered, scaled, and removed the effect of covariates.</p>
<pre class="r"><code># ON CRI
library(data.table)
library(readr)
library(Matrix)
library(susieR)
geno.file = &#39;height.ZBTB38.raw.gz&#39;
cat(&quot;Reading genotype data.\n&quot;)
geno &lt;- fread(geno.file,sep = &quot;\t&quot;,header = TRUE,stringsAsFactors = FALSE)
class(geno) &lt;- &quot;data.frame&quot;

# Extract the genotypes.
X &lt;- as(as.matrix(geno[-(1:6)]),&#39;dgCMatrix&#39;)

pheno.file &lt;- &quot;/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz&quot;

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat(&quot;Reading phenotype data.\n&quot;)
pheno        &lt;- suppressMessages(read_csv(pheno.file))
class(pheno) &lt;- &quot;data.frame&quot;
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread(&#39;height.ZBTB38.psam&#39;)
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20, data = pheno)

# Remove intercept
Z = Z[,-1]
Z = scale(Z, center=TRUE, scale=FALSE)
# standardize quantitative columns
cols = which(colnames(Z) %in% c(&quot;age&quot;,&quot;pc_genetic1&quot;,&quot;pc_genetic2&quot;,&quot;pc_genetic3&quot;,&quot;pc_genetic4&quot;,
                                &quot;pc_genetic5&quot;,&quot;pc_genetic6&quot;,&quot;pc_genetic7&quot;,&quot;pc_genetic8&quot;,&quot;pc_genetic9&quot;, 
                                &quot;pc_genetic10&quot;,&quot;pc_genetic11&quot;,&quot;pc_genetic12&quot;,&quot;pc_genetic13&quot;,&quot;pc_genetic14&quot;,
                                &quot;pc_genetic15&quot;,&quot;pc_genetic16&quot;,&quot;pc_genetic17&quot;,&quot;pc_genetic18&quot;,&quot;pc_genetic19&quot;,&quot;pc_genetic20&quot;))
Z[,cols] = scale(Z[,cols])
Z[,&#39;age2&#39;] = Z[,&#39;age&#39;]^2

# Center X
X = scale(X, center=TRUE, scale = FALSE)
xtxdiag = colSums(X^2)

A   &lt;- crossprod(Z)
# chol decomposition for (Z&#39;Z)^(-1)
R = chol(solve(A))
W = R %*% t(Z) %*% X

# Remove Covariates from X
X   &lt;- as.matrix(X - Z %*% crossprod(R,W))

# Get estimated parameters
ss.dat = readRDS(&#39;height.ZBTB38.XtX.Xty.rds&#39;)
betas = as.vector(ss.dat$Xty)/diag(ss.dat$XtX)
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)), standardize = T)
sigma2 = mod_bhat.ld$sigma2
b2 = c(mod_bhat.ld$mu[1, 1281], mod_bhat.ld$mu[2, 1312])/mod_bhat.ld$X_column_scale_factors[1281, 1312]
Xb = as.vector(X[, c(1281, 1312)] %*% b2)

n = nrow(X)
result = vector(&quot;list&quot;, 1000)
set.seed(201910)
for(i in 1:1000){
  ## generate y
  y = Xb + rnorm(n, 0, sqrt(sigma2))
  
  ## compute Xty, yty
  Xty = as.vector(y %*% X)
  yty = sum(y^2)
  
  ## compute summary stats
  betas = as.vector(Xty/diag(ss.dat$XtX))
  rss = yty - betas * Xty
  se = as.vector(sqrt(rss/((n-1)*diag(ss.dat$XtX))))
  z = betas/se
  
  result[[i]] = susie_bhat(betas, se, R=R, n=n, var_y = as.numeric(yty/(n-1)), standardize = T)
}
saveRDS(result, &#39;height.ZBTB38.simulation1000.rds&#39;)</code></pre>
<p>Load simulation results:</p>
<pre class="r"><code>result = readRDS(&#39;output/height.ZBTB38.simulation1000.rds&#39;)</code></pre>
<p>In 1000 simulations, 320 runs have 2 CSs. The rests have 1 CS.</p>
<pre class="r"><code>table(sapply(result, function(mod) length(mod$sets$cs)))</code></pre>
<pre><code>
  1   2 
680 320 </code></pre>
<p>291 runs contain both true signals.</p>
<pre class="r"><code>contain.true = sapply(result, function(mod) all(c(1281, 1312) %in% unlist(mod$sets$cs)))
sum(contain.true)</code></pre>
<pre><code>[1] 291</code></pre>
<p>963 runs contain rs2871960.</p>
<pre class="r"><code>contain.true.1 = sapply(result, function(mod) all( 1281 %in% unlist(mod$sets$cs)))
sum(contain.true.1)</code></pre>
<pre><code>[1] 963</code></pre>
<p>302 runs contain rs11919556.</p>
<pre class="r"><code>contain.true.2 = sapply(result, function(mod) all( 1312 %in% unlist(mod$sets$cs)))
sum(contain.true.2)</code></pre>
<pre><code>[1] 302</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.1.1     susieR_0.8.1.0545 gridExtra_2.3     dplyr_0.8.0.1    
[5] readr_1.3.1      

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       plyr_1.8.4       pillar_1.3.1     compiler_3.5.1  
 [5] later_0.7.5      git2r_0.26.1     workflowr_1.5.0  tools_3.5.1     
 [9] digest_0.6.18    evaluate_0.12    tibble_2.0.1     gtable_0.2.0    
[13] lattice_0.20-38  egg_0.4.5        pkgconfig_2.0.2  rlang_0.3.1     
[17] Matrix_1.2-15    yaml_2.2.0       withr_2.1.2      stringr_1.3.1   
[21] knitr_1.20       fs_1.3.1         hms_0.4.2        rprojroot_1.3-2 
[25] grid_3.5.1       tidyselect_0.2.5 glue_1.3.0       R6_2.3.0        
[29] rmarkdown_1.10   purrr_0.3.2      magrittr_1.5     whisker_0.3-2   
[33] backports_1.1.2  scales_1.0.0     promises_1.0.1   htmltools_0.3.6 
[37] assertthat_0.2.1 colorspace_1.4-0 httpuv_1.4.5     labeling_0.3    
[41] stringi_1.2.4    lazyeval_0.2.1   munsell_0.5.0    crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
