---
title: "Finemap Height"
author: "Yuxin Zou"
date: "7/20/2019"
output: 
  html_document:
    code_folding: hide
---

Load packages
```{r warning=FALSE, message=FALSE}
library(ggplot2)
library(susieR)
library(data.table)
```

We extract SNPs on chr 3 from 141,027,000 to 141,163,045. 
We choose this region based on result from http://big.stats.ox.ac.uk/pheno/biobank_50.

The code below are executed in CRI to extract SNPs.
```
zcat /gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz | awk 'BEGIN{FS=","} {print $1 OFS $1}' | tail -n +2 > height.id.txt

plink2 --sample /gpfs/data/xhe-lab/uk-biobank/data/genotypes/ukb27386_imp_v3_s487324.sample \
--bgen /gpfs/data/pierce-lab/uk-biobank-genotypes/ukb_imp_chr3_v3.bgen \
--chr 3 --from-bp 141027000 --to-bp 141163045 \
--keep height.id.txt \
--make-pgen --threads 8 --memory 38000 \
--out height.chr3 > height.chr3.plink2.log

plink2 --pfile height.chr3 --threads 8 --export A-transpose --out height.chr3

gzip height.chr3.traw
```

There are 4046 variantes in the extracted file.

Compute AF using plink2:
```
plink2 --pfile height.chr3 --freq --out height.chr3
```

Transform pgen file to bgen and compute LD matrix using LDstore:

Note `bgen-1.3` is not supported by LD store.
```
plink2 --pfile height.chr3 --recode bgen-1.2 --out height.chr3
ldstore --bgen height.chr3.bgen --bcor height.chr3.bcor --n-threads 10 --accuracy high --ld-thold 0.000001
ldstore --bcor height.chr3.bcor --merge 10
ldstore --bcor height.chr3.bcor --matrix height.chr3.matrix
```

Load genotype matrix in R and save the map and maf information:
```{r eval=FALSE}
library(data.table)
library(readr)
geno.file = 'height.chr3.traw.gz'
cat("Reading genotype data.\n")
geno <- fread(geno.file,sep = "\t",header = TRUE,stringsAsFactors = FALSE)
class(geno) <- "data.frame"

# Extract the SNP information.
map <- geno[1:6]

# Extract the genotypes.
geno <- geno[-(1:6)]
geno <- as.matrix(geno)
X = t(geno)
maf <- read.delim('height.chr3.afreq')
maf <- pmin(maf$ALT_FREQS, 1-maf$ALT_FREQS)
saveRDS(list(maf = maf, map=map), 'info.rds')
```

We remove the effect of 

1. sex
2. age and $age^2$
3. assessment_centre
4. genotype_measurement_batch
5. principal components 1â€“20

from **only height**.

```{r eval=FALSE}
pheno.file <- "/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz"

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat("Reading phenotype data.\n")
pheno        <- suppressMessages(read_csv(pheno.file))
class(pheno) <- "data.frame"
rownames(pheno) <- pheno$id
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20, data = pheno)
# delete intercept
Z = Z[,-1]
# standardize quantitative columns
cols = which(colnames(Z) %in% c("age","pc_genetic1","pc_genetic2","pc_genetic3","pc_genetic4",
                                "pc_genetic5","pc_genetic6","pc_genetic7","pc_genetic8","pc_genetic9", 
                                "pc_genetic10","pc_genetic11","pc_genetic12","pc_genetic13","pc_genetic14",
                                "pc_genetic15","pc_genetic16","pc_genetic17","pc_genetic18","pc_genetic19","pc_genetic20"))
Z[,cols] = scale(Z[,cols])
Z[,'age2'] = Z[,'age']^2

# Remove Covariates
y = pheno$height
A   <- crossprod(Z)
SZy <- solve(A,drop(y %*% Z))
y   <- y - drop(Z %*% SZy)
pheno.table = data.frame(FID = pheno$id, IID = pheno$id, height = y)
write.table(pheno.table, file = 'pheno.height.remove.sex.txt', quote = FALSE, row.names = FALSE) 
```

Compute sufficient statistics:
```{r eval=FALSE}
cat("Center data.\n")
X.c = scale(X, center = T, scale = F)
y.c = y - mean(y)

cat('Save data.\n')
saveRDS(list(XtX = crossprod(X.c), Xty = crossprod(X.c, y.c), yty = sum(y.c^2), n = length(y.c)), 'susie_ss_input_sex.rds')
```

Get summary statistics using plink:
```
plink2 --linear omit-ref no-x-sex --max-corr 1 --pfile height.chr3 --pheno pheno.height.remove.sex.txt --out height.chr3.remove.sex
```

```{r}
lm.res = read.delim('../data/height.chr3.remove.sex.height.glm.linear')
info = readRDS('../data/info.rds')
dat = readRDS('../data/susie_ss_input_sex.rds')
ld.matrix = as.matrix(fread('../data/height.chr3.matrix'))
```

Comparing summary statistics from PLINK with Neale's result, the estimated t statistics agree.

```{r eval=FALSE}
neale.file = '/gpfs/data/stephens-lab/finemap-uk-biobank/data/summary/neale/50_raw.gwas.imputed_v3.both_sexes.tsv.gz'
neale       <- suppressMessages(read_delim(neale.file, delim='\t'))
neale.pos = do.call(rbind, strsplit(neale$variant, "\\:"))
snps.pos = which(neale.pos[,2] %in% res$POS)
snps.chr = which(neale.pos[,1] == "3")
snps = intersect(snps.pos, snps.chr)
neale.choose = neale[snps,]
neale.choose$POS = as.numeric(neale.pos[snps,2])
saveRDS(neale.choose, 'height.chr3.neale.rds')
```

```{r}
neale.choose = readRDS('../data/height.chr3.neale.rds')
par(mfrow=c(1,2))
POS <- intersect(lm.res$POS, neale.choose$POS)
rows1 <- match(POS,lm.res$POS)
rows2 <- match(POS,neale.choose$POS)
{plot(neale.choose$pval[rows2], lm.res$P[rows1], xlab = 'Neale', ylab = 'result', main='p values')
abline(0,1)}
{par(pty='s')
plot(-log10(neale.choose$pval[rows2]), -log10(lm.res$P[rows1]), xlab = 'Neale', ylab = 'result', main='-log10(p values)', xlim=c(0,300), ylim=c(0,300))
abline(0,1)}
```

```{r}
par(pty='s')
plot(neale.choose$tstat[rows2], lm.res$T_STAT[rows1], xlab='Neale', ylab = 'result', main = 't statistics', xlim=c(-30,30), ylim=c(-30,30))
abline(0,-1)
```

We plot the p values from PLINK, and color points according to the correlations with rs2871960. The SNP rs2871960 is marked with diamond.

```{r}
tmp = data.frame(POS = lm.res$POS, p = -log10(lm.res$P), ref = lm.res$POS == 141121814, r2 = cut(ld.matrix[which(lm.res$POS == 141121814),]^2, c(0,0.2,0.4,0.6,0.8,1)))
ggplot(tmp, aes(x = POS, y = p, color=r2, shape = ref, size=ref)) + geom_point() + 
  scale_color_manual(name = expression(r^2),
                     values = c("(0,0.2]" = "darkblue", "(0.2,0.4]" = "deepskyblue", "(0.4,0.6]" = "lightgreen", "(0.6,0.8]" = "orange", "(0.8,1]" = "red"),
                     labels = c(expression(""<= 0.2), expression(paste("0.2<",r^2<=0.4)), expression(paste("0.4<",r^2<=0.6)), expression(paste("0.6<",r^2<=0.8)), expression(paste("0.8<",r^2<=1)))) +
  scale_shape_manual(values=c(20, 18), guide=FALSE) + scale_size_manual(values=c(2,5), guide=FALSE)
```

Fit susie model using sufficient statistics XtX, Xty:

```{r}
mod = susie_ss(XtX=dat$XtX, Xty = dat$Xty, yty = dat$yty, n = dat$n, maf = info$maf, maf_thresh = 0.01)
susie_plot(mod, y='PIP')
```
From the plot of bhat based on `XtX` and `Xty` vs PLINK result, it's clear that the bhat differs a lot!
```{r}
bhat = dat$Xty/diag(dat$XtX)
{plot(bhat, lm.res$BETA, xlab = 'Xty/diag(XtX)', ylab='PLINK')
abline(0,1)}
```

Fit susie bhat and rss model using `R = cov2cor(XtX)` and summary statistics from PLINK:
```{r warning=FALSE}
mod_bhat.R = susie_bhat(bhat = lm.res$BETA, shat = lm.res$SE, R = cov2cor(dat$XtX), n=dat$n, var_y = dat$yty/(dat$n - 1), maf = info$maf, maf_thresh = 0.01)
mod_rss.R = susie_rss(z = lm.res$T_STAT, R = cov2cor(dat$XtX), maf = info$maf, maf_thresh = 0.01)
{par(mfrow=c(1,2))
susie_plot(mod_bhat.R, y='PIP', main='SuSiE bhat XtX')
susie_plot(mod_rss.R, y='PIP', main='SuSiE rss XtX')
}
```

Fit susie bhat and rss model using LD matrix from LDstore and summary statistics from PLINK:
```{r}
mod_bhat.ld = susie_bhat(bhat = lm.res$BETA, shat = lm.res$SE, R = ld.matrix, n = dat$n, var_y = dat$yty/(dat$n - 1), maf=info$maf, maf_thresh = 0.01)
mod_rss.ld = susie_rss(z = lm.res$T_STAT, R = ld.matrix, maf=info$maf, maf_thresh = 0.01)
{par(mfrow=c(1,2))
susie_plot(mod_bhat.ld, y='PIP', main='SuSiE bhat LD')
susie_plot(mod_rss.ld, y='PIP', main='SuSiE rss LD')
}
```

The credible set contains 
```{r}
info$map[which(info$maf > 0.01),][unlist(mod_bhat.ld$sets$cs),]
```

