---
title: "Fine-mapping Height ZBTB38 1Mb"
author: "Yuxin Zou"
date: "8/15/2019"
output: 
  html_document:
    code_folding: hide
---

Load packages
```{r warning=FALSE, message=FALSE}
library(dplyr)
library(ggplot2)
library(susieR)
library(data.table)
library(Matrix)
library(gridExtra)
library(readr)
library(cowplot)
```

## Preparation

We extract SNPs on chr 3 from 140.621814 to 141.621814 with MAF > 0.01 and inputation score > 0.9. 
We choose this region based on result from http://big.stats.ox.ac.uk/pheno/biobank_50.

The code below are executed in CRI to extract SNPs. The code to get phenotype height is [code](scripts/get_pheno.R)

```
zcat /gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz | awk 'BEGIN{FS=","} {print $1 OFS $1}' | tail -n +2 > height.id.txt
zcat /gpfs/data/pierce-lab/uk-biobank-genotypes/ukb_mfi_chr3_v3.txt.gz | awk '($8 < 0.9){print $2}' > ukb_mfi_chr3_v3_exclude_id.txt

plink2 --sample /gpfs/data/xhe-lab/uk-biobank/data/genotypes/ukb27386_imp_v3_s487324.sample \
--bgen /gpfs/data/pierce-lab/uk-biobank-genotypes/ukb_imp_chr3_v3.bgen \
--chr 3 --from-bp 140800000 --to-bp 141800000 \
--keep height.id.txt \
--exclude ukb_mfi_chr3_v3_exclude_id.txt \
--maf 0.01 minor \
--snps-only --max-alleles 2 --rm-dup exclude-all --make-pgen --threads 8 --memory 38000 \
--out height.ZBTB38 > height.ZBTB38.plink2.log

plink2 --pfile height.ZBTB38 --threads 8 --export A --out height.ZBTB38

gzip height.ZBTB38.raw
```

There are 2701 variantes in the extracted file.

Compute AF using plink2:
```
plink2 --pfile height.ZBTB38 --freq --out height.ZBTB38
```

Transform pgen file to bgen and compute LD matrix using LDstore:

Note that `bgen-1.3` is not supported by LD store.
```
plink2 --pfile height.ZBTB38 --recode bgen-1.2 --out height.ZBTB38

ldstore --bgen height.ZBTB38.bgen --bcor height.ZBTB38.bcor --n-threads 10 --accuracy high --ld-thold 0
ldstore --bcor height.ZBTB38.bcor --merge 10
ldstore --bcor height.ZBTB38.bcor --matrix height.ZBTB38.matrix
rm height.${regionname}.bcor_*
```

Load genotype matrix in R:
```{r eval=FALSE}
library(data.table)
library(readr)
library(Matrix)
geno.file = 'height.ZBTB38.raw.gz'
cat("Reading genotype data.\n")
geno <- fread(geno.file,sep = "\t",header = TRUE,stringsAsFactors = FALSE)
class(geno) <- "data.frame"

# Extract the genotypes.
X <- as(as.matrix(geno[-(1:6)]),'dgCMatrix')
```

We remove the effect of 

1. sex
2. age and $age^2$
3. assessment_centre
4. genotype_measurement_batch
5. principal components 1â€“20

from both genotype matrix and height.

```{r eval=FALSE}
pheno.file <- "/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz"
out.pheno.file <- "pheno.height.txt"
out.covar.file <- "covar.height.txt"

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat("Reading phenotype data.\n")
pheno        <- suppressMessages(read_csv(pheno.file))
class(pheno) <- "data.frame"
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread('height.ZBTB38.psam')
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20, data = pheno)

# Remove intercept
Z = Z[,-1]
Z = scale(Z, center=TRUE, scale=FALSE)
# standardize quantitative columns
cols = which(colnames(Z) %in% c("age","pc_genetic1","pc_genetic2","pc_genetic3","pc_genetic4",
                                "pc_genetic5","pc_genetic6","pc_genetic7","pc_genetic8","pc_genetic9", 
                                "pc_genetic10","pc_genetic11","pc_genetic12","pc_genetic13","pc_genetic14",
                                "pc_genetic15","pc_genetic16","pc_genetic17","pc_genetic18","pc_genetic19","pc_genetic20"))
Z[,cols] = scale(Z[,cols])
Z[,'age2'] = Z[,'age']^2

# Save phenotype file
pheno.table = data.frame(FID = pheno$id, IID = pheno$id, height = pheno$height)
write.table(pheno.table, file = out.pheno.file, quote = FALSE, row.names = FALSE)

# Save covariates file
IID = pheno$id
FID = pheno$id
cov.table = cbind(FID, IID, Z)
write.table(cov.table, file= out.covar.file, quote=FALSE, row.names = FALSE)

# Remove Covariates from y
y = pheno$height
names(y) = pheno$id
# Center y
y = y - mean(y)
A   <- crossprod(Z)
SZy <- solve(A,drop(y %*% Z))
y   <- y - drop(Z %*% SZy)

# Center X
X = scale(X, center=TRUE, scale = FALSE)
xtxdiag = colSums(X^2)

# chol decomposition for (Z'Z)^(-1)
R = chol(solve(A))
W = R %*% t(Z) %*% X

# Remove Covariates from X
X   <- as.matrix(X - Z %*% crossprod(R,W))

# Load LD matrix from raw genotype
ld.matrix = as.matrix(fread('height.ZBTB38.matrix'))
# X'X
XtX = sqrt(xtxdiag) * t(ld.matrix*sqrt(xtxdiag)) - crossprod(W)
rownames(XtX) = colnames(XtX) = colnames(X)
# X'y
Xty = as.vector(y %*% X)

## SNP info
maf <- read.delim('height.ZBTB38.afreq')
pos <- fread('height.ZBTB38.pvar')
pos$maf = pmin(maf$ALT_FREQS, 1-maf$ALT_FREQS)

## Save results
save(X, y, pos, file = 'height.ZBTB38.X.y.RData')
saveRDS(list(XtX = XtX, Xty = Xty, yty = sum(y^2), n = length(y), pos=pos), 'height.ZBTB38.XtX.Xty.rds')
```

Get summary statistics:
```{r}
ss.dat = readRDS('data/height.ZBTB38.XtX.Xty.rds')
betas = ss.dat$Xty/diag(ss.dat$XtX)
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX)))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
```

Get gene data
```{r}
genes        <- read_delim("data/seq_gene.md.gz",delim = "\t",quote = "")
class(genes) <- "data.frame"
genes        <- subset(genes,
                       group_label == "GRCh37.p5-Primary Assembly" &
                       feature_type == "GENE")
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == 3 &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
```

## Check summary stats

Comparing with result from plink, the estimated t statistics agree.
The PLINK command to get the summary statistics:
```
plink2 --linear hide-covar no-x-sex omit-ref --pfile height.ZBTB38 --covar covar.height.txt --pheno pheno.height.txt --vif 100 --out height.ZBTB38.plink2
```
```{r}
lm.res = read.delim('output/height.ZBTB38.plink2.height.glm.linear')
POS <- intersect(ss.dat$pos$POS, lm.res$POS)
rows1 <- match(POS,ss.dat$pos$POS)
rows2 <- match(POS,lm.res$POS)
{par(pty='s')
plot(-log10(lm.res$P[rows2]), -log10(pval[rows1]), xlab = 'PLINK2', ylab = 'result', main='-log10(p values)', xlim=c(0,250), ylim=c(0,250))
abline(0,1)}
```

```{r}
par(pty='s')
plot(lm.res$T_STAT[rows2], z[rows1], xlab='PLINK2', ylab = 'result', main = 't statistics', xlim=c(-30,30), ylim=c(-30,30))
abline(0,-1)
```

Comparing summary statistics with Neale's result, the results agree well.

```{r, fig.align='center', fig.height=5, fig.width=10}
map = fread(paste0('data/region-variants-ZBTB38.csv.gz'))
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = -log10(neale_pval),y = -(pnorm(-abs(z), log.p = T) + log(2))/log(10))) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "-log10(p value)")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
grid.arrange(p1,p2, ncol=2)
```

We plot the p values from PLINK, and color points according to the correlations with rs2871960. The SNP rs2871960 is marked with diamond.

```{r}
## Plot functions
#' plot gene name annotations
#' @param dat a matrix of gene names with 'start' and 'end' base-pair position
#' @param xrange range of x axis, base-pair position
plot_geneName = function(dat, xrange){
  ngene = 2:nrow(dat)
  line = 1
  dat$lines = NA
  dat$lines[1] = 1
  gene.end = dat[1, 'end']
  while(length(ngene) != 0){
    id = which(dat[ngene, 'start'] > gene.end + 0.02)[1]
    if(!is.na(id)){
      dat$lines[ngene[id]] = line
      gene.end = dat[ngene[id],'end']
      ngene = ngene[-id]
    }else{
      line = line + 1
      dat$lines[ngene[1]] = line
      gene.end = dat[ngene[1],'end']
      ngene = ngene[-1]
    }
  }
  
  dat$mean = rowMeans(dat[,c('start', 'end')])
  
  pl = ggplot(dat, aes(xmin = xrange[1], xmax = xrange[2], ymin = -lines-0.2)) + 
    geom_rect(aes(xmin = start, xmax = end, ymin = -lines-0.05, ymax = -lines+0.05), fill='blue') +
    geom_text(aes(x = mean, y=-lines-0.3, label=geneName), size=2.5) + xlab('base-pair position') + ylab('Gene') + 
    theme_bw() + theme(axis.text.x=element_blank(),
                       axis.ticks = element_blank(),
                       axis.text.y=element_blank(),
                       plot.title=element_text(size=11),
                       panel.grid.major = element_blank(),
                       panel.grid.minor = element_blank())
  pl
}

discrete_gradient_pal <- function(colours, bins = 5) {
  ramp <- scales::colour_ramp(colours)
  
  function(x) {
    if (length(x) == 0) return(character())
    
    i <- floor(x * bins)
    i <- ifelse(i > bins-1, bins-1, i)
    ramp(i/(bins-1))
  }
}

scale_colour_discrete_gradient <- function(..., colours, bins = 5, na.value = "grey50", guide = "colourbar", aesthetics = "colour", colors)  {
  colours <- if (missing(colours)) 
    colors
  else colours
  continuous_scale(
    aesthetics,
    "discrete_gradient",
    discrete_gradient_pal(colours, bins),
    na.value = na.value,
    guide = guide,
    ...
  )
}

#' Locuszoom plot
#' @param z a vector of z scores with SNP names
#' @param pos base-pair positions
#' @param gene.pos.map a matrix of gene names with 'start' and 'end' base-pair position
#' @param z.ref.name the reference SNP
#' @param ld correlations between teh reference SNP and the rests
#' @param title title of the plot
#' @param title.size the size of the title
#' @param true the true value
#' @param y.height height of -log10(p) plot and height of the gene name annotation plot
#' @param y.lim range of y axis
locus.zoom = function(z, pos, gene.pos.map=NULL, z.ref.name=NULL, ld=NULL, 
                      title = NULL, title.size = 7, true = NULL, 
                      y.height=c(5,1.5), y.lim=NULL){
  tmp = data.frame(POS = pos, p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10))
  pl_zoom = ggplot(tmp, aes(x = POS, y = p)) + geom_point(color = 'darkblue') + 
    ylab("-log10(p value)") + ggtitle(title) + 
    theme_bw() + theme(axis.title.x=element_blank(),
                       plot.title = element_text(size=title.size))
  if(!is.null(ld) && !is.null(z.ref.name)){
    tmp$ref = names(z) == z.ref.name
    tmp$r2 = ld^2
    pl_zoom = ggplot(tmp, aes(x = POS, y = p, shape = ref, size=ref, color=r2)) + geom_point() + 
      ylab("-log10(p)") + ggtitle(title) + 
      scale_colour_discrete_gradient(
        colours = c("darkblue", "deepskyblue", "lightgreen", "orange", "red"),
        limits = c(0, 1.01),
        breaks = c(0,0.2,0.4,0.6,0.8,1),
        guide = guide_colourbar(nbin = 100, raster = FALSE, frame.colour = "black", ticks.colour = NA)
      ) + 
      scale_shape_manual(values=c(20, 18), guide=FALSE) + scale_size_manual(values=c(2,5), guide=FALSE) + 
      theme_bw() + theme(axis.title.x=element_blank(),
                         plot.title = element_text(size=title.size))
  }
  
  if(!is.null(y.lim)){
    pl_zoom = pl_zoom + ylim(y.lim[1], y.lim[2])
  }
  pl_zoom = pl_zoom + geom_hline(yintercept=-log10(5e-08), linetype='dashed', color = 'red')
  if(!is.null(true)){
    tmp.true = data.frame(POS = which(true!=0), p = tmp$p[which(true!=0)], 
                          ref = (names(z) == z.ref.name)[which(true!=0)],
                          label = paste0('SNP',1:length(which(true!=0))))
    pl_zoom = pl_zoom + geom_point(data=tmp.true, aes(x=POS, y=p), 
                                   color='red', show.legend = FALSE, shape=1, stroke = 1) + 
      geom_text(data=tmp.true, aes(x = POS-30, y=p+1, label=label), size=3, color='red')
  }
  if(!is.null(gene.pos.map)){
    pl_gene = plot_geneName(gene.pos.map, xrange = c(min(pos), max(pos)))
    g = egg::ggarrange(pl_zoom, pl_gene, nrow=2, heights = y.height, draw=FALSE)
  }else{
    g = pl_zoom
  }
  g
}

#' SuSiE plot with Locuszoom plot
#' @param z a vector of z scores with SNP names
#' @param model the fitted SuSiE model
#' @param pos base-pair positions
#' @param gene.pos.map a matrix of gene names with 'start' and 'end' base-pair position
#' @param z.ref.name the reference SNP
#' @param ld correlations between teh reference SNP and the rests
#' @param title title of the plot
#' @param title.size the size of the title
#' @param true the true value
#' @param plotz whether to plot locuszoom plot
#' @param y.lim range of y axis
#' @param y.susie the y axis of the SuSiE plot, 'PIP' or 'p', 'p' refers to -log10(p)
susie_plot_locuszoom = function(z, model, pos, gene.pos.map = NULL, z.ref.name, ld, 
                                title = NULL, title.size = 7, true = NULL, 
                                plotz = TRUE, y.lim=NULL, y.susie='PIP'){
  if(plotz){
    pl_zoom = locus.zoom(z, pos = pos, ld=ld, z.ref.name = z.ref.name, title = title, title.size = title.size, y.lim=y.lim)
  }
  pip = model$pip
  tmp = data.frame(POS = pos, PIP = pip, p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10))
  if(y.susie == 'PIP'){
    if(plotz){
      pl_susie = ggplot(tmp, aes(x = POS, y = PIP)) + geom_point(show.legend = FALSE, size=3) + 
        theme_bw() + theme(axis.title.x=element_blank(), axis.text.x=element_blank())
    }else{
      pl_susie = ggplot(tmp, aes(x = POS, y = PIP)) + geom_point(show.legend = FALSE, size=3) + 
        theme_bw() + theme(axis.title.x=element_blank())
    }
  }else if(y.susie == 'p'){
    if(plotz){
      pl_susie = ggplot(tmp, aes(x = POS, y = p)) + geom_point(show.legend = FALSE, size=3) + 
        ylab("-log10(p)") + 
        theme_bw() + theme(axis.title.x=element_blank(), axis.text.x=element_blank())
    }else{
      pl_susie = ggplot(tmp, aes(x = POS, y = p)) + geom_point(show.legend = FALSE, size=3) +
        ylab("-log10(p)") + 
        theme_bw() + theme(axis.title.x=element_blank())
    }
    
  }
  
  if(!is.null(true)){
    tmp.true = data.frame(POS = pos[which(true!=0)], PIP = pip[which(true!=0)])
    pl_susie = pl_susie + geom_point(data=tmp.true, aes(x=POS, y=PIP), 
                                     color='red', size=3, show.legend = FALSE)
  }
  
  model.cs = model$sets$cs
  if(!is.null(model.cs)){
    for(i in 1:length(model.cs)){
      tmp.cs = tmp[model.cs[[i]],]
      if(y.susie == 'PIP'){
        pl_susie = pl_susie + geom_point(data=tmp.cs, aes(x=POS, y=PIP), 
                                       color = i+4, size=3, show.legend = FALSE, shape=1, stroke = 1)
      }else if(y.susie == 'p'){
        pl_susie = pl_susie + geom_point(data=tmp.cs, aes(x=POS, y=p), 
                                       color = i+4, size=3, show.legend = FALSE, shape=1, stroke = 1)
      }
    }
  }
  
  if(!is.null(gene.pos.map)){
    pl_gene = plot_geneName(gene.pos.map, xrange = c(min(pos), max(pos)))
    if(plotz){
      g = egg::ggarrange(pl_zoom, pl_susie, pl_gene, nrow=3, heights = c(4,4,1.5), draw=FALSE)
    }else{
      g = egg::ggarrange(pl_susie, pl_gene, nrow=2, heights = c(4,2), draw=FALSE)
    }
    
  }else{
    if(plotz){
      g = egg::ggarrange(pl_zoom, pl_susie, nrow=2, heights = c(4,4), draw=FALSE)
    }else{
      g = pl_susie
    }
  }
  g
}
```

```{r fig.width=15}
locus.zoom(z, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C')
```

## SuSiE

Fit susie model using individual level data:

The following code is submited on CRI.
```{r eval=FALSE}
library(susieR)
library(Matrix)
load('height.ZBTB38.X.y.RData')

X = as.matrix(X)
time_mod = system.time(mod <- susie(X=X, Y=y, verbose=T, standardize = FALSE))
cat(sprintf("Fitting SuSiE model tooks %d seconds.\n",round(time_mod["elapsed"])))

saveRDS(mod,'height.ZBTB38.susie.model.rds')
```

Result:
```
Resource List:     mem=30gb,walltime=08:00:00,nodes=1:ppn=10,neednodes=1:ppn=10
Resources Used:    cput=00:32:43,vmem=35429564kb,walltime=00:39:23,mem=31456804kb,energy_used=0

Fitting SuSiE model tooks 2304 seconds.
```

```{r fig.align='center', fig.width=30, fig.height=15}
mod = readRDS('output/height.ZBTB38.susie.model.rds')
p1 = susie_plot_locuszoom(z, mod, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C')
p2 = susie_plot_locuszoom(z, mod, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C', y.susie ='p')
grid.arrange(p1, p2, ncol=2)
```

Fit susie bhat and rss model using LD matrix from LDstore and summary statistics from PLINK:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = as.vector(betas), shat = as.vector(se), R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty)/(ss.dat$n-1), standardize = FALSE)
mod_bhat.ld.stand = susie_bhat(bhat = as.vector(betas), shat = as.vector(se), R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty)/(ss.dat$n-1), standardize = TRUE)
mod_rss.ld = susie_rss(z = as.vector(z), R = R)
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C', title = 'SuSiE bhat without standardize')
p_bhat_stand = susie_plot_locuszoom(z, mod_bhat.ld.stand, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C', title = 'SuSiE bhat with standardize')
p_rss = susie_plot_locuszoom(z, mod_rss.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C', title = 'SuSiE RSS')
grid.arrange(p_bhat, p_bhat_stand, p_rss, ncol=2)
```

The first credible set contains 
```{r}
cs1 = ss.dat$pos[unlist(mod_bhat.ld$sets$cs$L1),]
cs1$gene = sapply(cs1$POS/1e6, function(i){
  id = intersect(which(gene.pos.map$start <= i), which(gene.pos.map$end >= i))
  gene.pos.map$geneName[id]
})
cs1
```


