---
title: "Fine-mapping Height ZBTB38 -- SuSiE check"
author: "Yuxin Zou"
date: "10/07/2019"
output: 
  html_document:
    code_folding: hide
---

We perform some check for the SuSiE result on region around ZBTB38.

Load pacakges:
```{r message=FALSE}
library(readr)
library(dplyr)
library(gridExtra)
```
Load plotting functions:
```{r}
knitr::read_chunk("../scripts/plots.R")
```

```{r plots}
```

```{r}
locus.zoom.cs = function(z, cs, pos, gene.pos.map=NULL, z.ref.name=NULL, ld=NULL, title = NULL, title.size = 10, xrange = NULL, y.lab=NULL){
  if(is.null(y.lab)){
    y.lab = '-log10(p value)'
  }
  tmp = data.frame(POS = pos, log10p = -(pnorm(-abs(z), log.p = T) + log(2))/log(10))
  tmp$ref = names(z) == z.ref.name
  tmp$r2 = ld^2
  tmp$CS = rep(4, length(z))
  tmp$CS[cs] = 16
  tmp$CS = as.factor(tmp$CS)
  pl_zoom = ggplot(tmp, aes(x = POS, y = log10p, shape = CS, color = r2, size=CS)) + 
    geom_point() + ylab(y.lab) + 
    scale_colour_discrete_gradient(
        colours = c("darkblue", "deepskyblue", "lightgreen", "orange", "red"),
        limits = c(0, 1.01),
        breaks = c(0,0.2,0.4,0.6,0.8,1),
        guide = guide_colourbar(nbin = 100, raster = FALSE, frame.colour = "black", ticks.colour = NA)) +
    scale_shape_manual(values = c(4, 16), guide=FALSE) + 
    scale_size_manual(values=c(1.5,4), guide=FALSE) + 
    ggtitle(title) + 
    theme_bw() + theme(axis.title.x=element_blank(), plot.title = element_text(size=title.size))
  tmp.sub = tmp[cs,]
  pl_zoom = pl_zoom + geom_point(data = tmp.sub, aes(x=POS, y=log10p), shape=1, size=4, color='black', stroke=0.1)
  if(!is.null(xrange)){
    pl_zoom = pl_zoom + xlim(xrange[1], xrange[2])
  }
  
  pl_gene = plot_geneName(gene.pos.map, xrange = xrange)
  g = egg::ggarrange(pl_zoom, pl_gene, nrow=2, heights = c(5,1.5), draw=FALSE)
  g
}
```

Get summary statistics:
```{r}
ss.dat = readRDS('../data/height.ZBTB38.0.01.XtX.Xty.rds')
betas = ss.dat$Xty/diag(ss.dat$XtX)
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX)))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
```

Get gene data:
```{r warning=FALSE, message=FALSE}
genes        <- read_delim("../data/seq_gene.md.gz",delim = "\t",quote = "")
class(genes) <- "data.frame"
genes        <- subset(genes,
                       group_label == "GRCh37.p5-Primary Assembly" &
                       feature_type == "GENE")
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == 3 &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(4, 9, 10, 15),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod = readRDS('../data/height.chr3.140800000.141800000.0.01.susie.model.rds')
p1 = susie_plot_locuszoom(z, mod, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C')
p2 = susie_plot_locuszoom(z, mod, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[which(ss.dat$pos$POS == 141121814),], z.ref.name = 'rs2871960_C', y.susie ='p')
grid.arrange(p1, p2, ncol=2)
```

For 1Mb region about ZBTB38, SuSiE found 2 CSs. The SNP with the strongest marginal p value in CS1 is rs2871960 (p = 7.200e-207). For CS2, the SNP with the strongest marginal p value is rs11919556 (p = 0.0213).

The correlation between rs2871960 and rs11919556 is `r R[945,978]`. The average correlation between SNPs in CS1 and CS2 is
```{r}
round(mean(abs(R[mod$sets$cs$L1, mod$sets$cs$L2])), 4)
```

## Removing effect of CS1

CS1:

```{r warning=FALSE}
z.max = which.max(abs(z))
locus.zoom.cs(z, cs = mod$sets$cs$L1, ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), xrange=c(140.9,141.5), title = 'ZBTB38 CS1')
```

```{r eval=FALSE}
library(data.table)
library(readr)
library(Matrix)
geno.file = 'height.ZBTB38.0.01.raw.gz'
cat("Reading genotype data.\n")
geno <- fread(geno.file,sep = "\t",header = TRUE,stringsAsFactors = FALSE)
class(geno) <- "data.frame"

# Extract the genotypes.
X <- as(as.matrix(geno[-(1:6)]),'dgCMatrix')

pheno.file <- "/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz"
out.pheno.file <- "pheno.height.txt"
out.covar.file <- "covar.removeCS1.height.txt"

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat("Reading phenotype data.\n")
pheno        <- suppressMessages(read_csv(pheno.file))
class(pheno) <- "data.frame"
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread('height.ZBTB38.0.01.psam')
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20 + X[,945], data = pheno)

# Remove intercept
Z = Z[,-1]
colnames(Z)[150] = 'X945'
# standardize quantitative columns
cols = which(colnames(Z) %in% c("age","pc_genetic1","pc_genetic2","pc_genetic3","pc_genetic4",
                                "pc_genetic5","pc_genetic6","pc_genetic7","pc_genetic8","pc_genetic9", 
                                "pc_genetic10","pc_genetic11","pc_genetic12","pc_genetic13","pc_genetic14",
                                "pc_genetic15","pc_genetic16","pc_genetic17","pc_genetic18","pc_genetic19","pc_genetic20", "X945"))
Z[,cols] = scale(Z[,cols])
Z[,'age2'] = Z[,'age']^2
# Save phenotype file
pheno.table = data.frame(FID = pheno$id, IID = pheno$id, height = pheno$height)
write.table(pheno.table, file = out.pheno.file, quote = FALSE, row.names = FALSE)
# Save covariates file
IID = pheno$id
FID = pheno$id
cov.table = cbind(FID, IID, Z)
write.table(cov.table, file= out.covar.file, quote=FALSE, row.names = FALSE)
```

```
plink2 --linear hide-covar no-x-sex omit-ref --pfile height.ZBTB38.0.01 --covar covar.removeCS1.height.txt --pheno pheno.height.txt --vif 100 --out height.ZBTB38.0.01.removeCS1.plink2
```

After removing the effect of rs2871960, the conditional p value for rs11919556 becomes 4.289e-06, and it becomes the strongest one among all SNPs!
```{r warning=FALSE}
lm.res = read.delim('../data/height.ZBTB38.0.01.removeCS1.plink2.height.glm.linear')
lm.res.t = lm.res$T_STAT
names(lm.res.t) = lm.res$ID
z.max = which.max(abs(lm.res.t))
locus.zoom.cs(lm.res.t, cs = mod$sets$cs$L2, pos=lm.res$POS/1e6, gene.pos.map = gene.pos.map, z.ref.name = lm.res$ID[z.max], ld = R[z.max,], xrange=c(140.9,141.5), title='ZBTB38 CS2', y.lab = '-log10(p value condition on CS1)')
```

## Simulation under the estimated model

In the following simulation, we treat rs2871960 and rs11919556 as true signals. The effect sizes are simulated using the estimated prior variance. We use the fitted residual variance in the simulation. The response y is simulated from
$$
y \sim N_n(Xb, \sigma^2 I)
$$
, where X the genotype matrix that column centered, scaled, and removed the effect of covariates.

```{r eval=FALSE}
# ON CRI
library(data.table)
library(readr)
library(Matrix)
library(susieR)
geno.file = 'height.ZBTB38.0.01.raw.gz'
cat("Reading genotype data.\n")
geno <- fread(geno.file,sep = "\t",header = TRUE,stringsAsFactors = FALSE)
class(geno) <- "data.frame"

# Extract the genotypes.
X <- as(as.matrix(geno[-(1:6)]),'dgCMatrix')

pheno.file <- "/gpfs/data/stephens-lab/finemap-uk-biobank/data/raw/height.csv.gz"

# LOAD PHENOTYPE and COVARIATES DATA
# -------------------
# Read the phenotype data from the CSV file.
cat("Reading phenotype data.\n")
pheno        <- suppressMessages(read_csv(pheno.file))
class(pheno) <- "data.frame"
pheno$sex = factor(pheno$sex)
pheno$assessment_centre = factor(pheno$assessment_centre)
pheno$genotype_measurement_batch = factor(pheno$genotype_measurement_batch)
pheno$age2 = pheno$age^2
# match individual order with genotype file
ind = fread('height.ZBTB38.0.01.psam')
match.idx = match(ind$IID, pheno$id)
pheno = pheno[match.idx,]

Z = model.matrix(~ sex + age + age2 + assessment_centre + genotype_measurement_batch +
                   pc_genetic1 + pc_genetic2 + pc_genetic3 + pc_genetic4 + pc_genetic5 +
                   pc_genetic6 + pc_genetic7 + pc_genetic8 + pc_genetic9 + pc_genetic10 +
                   pc_genetic11 + pc_genetic12 + pc_genetic13 + pc_genetic14 + pc_genetic15 +
                   pc_genetic16 + pc_genetic17 + pc_genetic18 + pc_genetic19 + pc_genetic20, data = pheno)

# Remove intercept
Z = Z[,-1]
# standardize quantitative columns
cols = which(colnames(Z) %in% c("age","pc_genetic1","pc_genetic2","pc_genetic3","pc_genetic4",
                                "pc_genetic5","pc_genetic6","pc_genetic7","pc_genetic8","pc_genetic9", 
                                "pc_genetic10","pc_genetic11","pc_genetic12","pc_genetic13","pc_genetic14",
                                "pc_genetic15","pc_genetic16","pc_genetic17","pc_genetic18","pc_genetic19","pc_genetic20"))
Z[,cols] = scale(Z[,cols])
Z[,'age2'] = Z[,'age']^2

# Center scale X
cm = Matrix::colMeans(X, na.rm = TRUE)
csd = susieR:::compute_colSds(X)
csd[csd == 0] = 1
X = t((t(X) - cm) / csd)

A   <- crossprod(Z)
# chol decomposition for (Z'Z)^(-1)
R = chol(solve(A))
W = R %*% t(Z) %*% X

# Remove Covariates from X
X   <- as.matrix(X - Z %*% crossprod(R,W))

# Get estimated parameters
mod = readRDS('height.chr3.140800000.141800000.0.01.susie.model.rds')
ss.dat = readRDS('height.ZBTB38.0.01.XtX.Xty.rds')
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
sigma2 = mod$sigma2
V = mod$V[1:2]
n = nrow(X)

result = vector("list", 1000)
set.seed(201910)
for(i in 1:1000){
  ## generate true effects
  b = c(rnorm(1, 0, sqrt(V[1])), rnorm(1, 0, sqrt(V[2])))
  Xb = as.vector(X[, c(945, 978)] %*% b)
  ## generate y
  y = Xb + rnorm(n, 0, sqrt(sigma2))
  
  ## compute Xty, yty
  Xty = as.vector(y %*% X)
  yty = sum(y^2)
  
  ## compute summary stats
  betas = as.vector(Xty/diag(ss.dat$XtX))
  rss = yty - betas * Xty
  se = as.vector(sqrt(rss/((n-1)*diag(ss.dat$XtX))))
  z = betas/se
  
  result[[i]] = susie_bhat(betas, se, R=R, n=n, var_y = as.numeric(yty/(n-1)))
}
saveRDS(result, 'height.ZBTB38.0.01.simulation1000.rds')
```

Load simulation results:
```{r}
result = readRDS('../data/height.ZBTB38.0.01.simulation1000.rds')
```

In 1000 runs, the number of CSs are 
```{r}
table(sapply(result, function(mod) length(mod$sets$cs)))
```

809 runs contain rs2871960 in the CSs, 250 runs contain rs11919556 in the CSs.
```{r}
contain.true = sapply(result, function(mod) 945 %in% unlist(mod$sets$cs))
sum(contain.true)
contain.true = sapply(result, function(mod) 978 %in% unlist(mod$sets$cs))
sum(contain.true)
```










