---
title: "Fine-mapping Height on More Regions"
author: "Yuxin Zou"
date: "10/06/2019"
output: 
  html_document:
    code_folding: hide
---

Load packages
```{r warning=FALSE, message=FALSE}
library(dplyr)
library(ggplot2)
library(susieR)
library(data.table)
library(Matrix)
library(gridExtra)
library(readr)
library(cowplot)
```

```{r}
knitr::read_chunk("../scripts/plots.R")
```

```{r plots}
```

We extract SNPs in 10 distinct regions with MAF > 0.01. 
We choose this region based on result from http://big.stats.ox.ac.uk/pheno/biobank_50.

| Region   | CHR | FROM (Mb) | TO (Mb) |
|----------|-----|----------:|--------:|
| EFEMP1   | 2   | 55.6      | 56.6    |
| ZBTB38   | 3   | 140.8     | 141.8   |
| LCORL    | 4   | 17.5      | 18.5    |
| HHIP     | 4   | 145       | 146     |
| GNA12    | 7   | 2.3       | 3.3     |
| CDK6     | 7   | 91.8      | 92.8    |
| HMGA2    | 12  | 65.6      | 66.6    |
| DLEU1    | 13  | 50.6      | 51.6    |
| ADAMTS17 | 15  | 100.2     | 101.2   |
| UQCC1    | 20  | 33.5      | 34.5    |

The code to get phenotype height is [get_height](../scripts/get_pheno.R). The code to compute summary statistics is in [prepare](../scripts/prepare.region.sh).

Get gene data
```{r warning=FALSE, message=FALSE}
genes        <- read_delim("../data/seq_gene.md.gz",delim = "\t",quote = "")
class(genes) <- "data.frame"
genes        <- subset(genes,
                       group_label == "GRCh37.p5-Primary Assembly" &
                       feature_type == "GENE")
```

## EFEMP1

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.EFEMP1")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-EFEMP1.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-10,]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'EFEMP1 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'EFEMP1 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## ZBTB38

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.ZBTB38.*XtX")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-ZBTB38.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(4, 9, 10, 15),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'ZBTB38 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'ZBTB38 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## LCORL

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.LCORL.*XtX")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-LCORL.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(3, 10),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'LCORL SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'LCORL SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## HHIP

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.HHIP")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-HHIP.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(3, 5),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'HHIP SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'HHIP SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## GNA12

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.GNA12")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-GNA12.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(3,4,6,15),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'GNA12 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'GNA12 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## CDK6

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.CDK6")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-CDK6.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(1,5,11),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'CDK6 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'CDK6 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## HMGA2

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.HMGA2")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-HMGA2.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(3,4,8),]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'HMGA2 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'HMGA2 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## DLEU1

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.DLEU1")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-DLEU1.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'DLEU1 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'DLEU1 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## ADAMTS17

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.ADAMTS17")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-ADAMTS17.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
gene.pos.map = gene.pos.map[-c(4, 5, 12), ]
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'ADAMTS17 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'ADAMTS17 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```

## UQCC1

Get summary statistics:
```{r}
file = dir("../data/", pattern="height.UQCC1")
ss.dat = readRDS(paste0('../data/', file))
betas = as.vector(ss.dat$Xty/diag(ss.dat$XtX))
rss = c(ss.dat$yty) - betas * ss.dat$Xty
se = as.vector(sqrt(rss/((ss.dat$n-1)*diag(ss.dat$XtX))))
z = betas/se
pval = 2*pnorm(-abs(z))
R = as.matrix(t(ss.dat$XtX * (1/sqrt(diag(ss.dat$XtX)))) * (1/ sqrt(diag(ss.dat$XtX))))
names(z) = rownames(R)
print(paste0(length(z), ' SNPs included'))
```

Comparing with Neale and GeneATLAS results:
```{r fig.align='center', fig.height=5, fig.width=10}
map = fread('../data/region-variants-UQCC1.csv')
z.table = data.frame('id' = gsub('_.*$', '', names(z)), 'z' = z)
map = merge(map, z.table, by='id', all = TRUE)
p1 <- ggplot(map,aes(x = NBETA/NSE,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "GeneATLAS",y = "Our result",title = "t-statistics")
p2 <- ggplot(map,aes(x = neale_tstat,y = z)) +
  geom_point(shape = 20,size = 2,na.rm = TRUE) +
  geom_abline(intercept = 0,slope = 1,color = "dodgerblue",
              linetype = "dotted") +
  xlim(c(-50,50)) +
  ylim(c(-50,50)) +
  theme_cowplot(font_size = 12) +
  labs(x = "Neale lab",y = "Our result",title = "t-statistics")
gridExtra::grid.arrange(p1, p2, ncol=2)
```

```{r}
start.pos <- min(ss.dat$pos$POS)
stop.pos  <- max(ss.dat$pos$POS)
plot.genes <- subset(genes,
                     chromosome == unique(ss.dat$pos$`#CHROM`) &
                     ((chr_start > start.pos & chr_start < stop.pos) |
                      (chr_stop > start.pos & chr_start < stop.pos)) & feature_type == 'GENE')
gene.pos.map = plot.genes %>% select(feature_name, chr_start, chr_stop)
colnames(gene.pos.map) = c('geneName', 'start', 'end')
gene.pos.map = as.data.frame(gene.pos.map)
gene.pos.map = gene.pos.map %>% mutate(start = start/1e6, end = end/1e6)
```

SuSiE result:
```{r fig.align='center', fig.width=30, fig.height=15}
mod_bhat.ld = susie_bhat(bhat = betas, shat = se, R = R, n = ss.dat$n, var_y = as.numeric(ss.dat$yty/(ss.dat$n-1)))
z.max = which.max(abs(z))
pip_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'UQCC1 SuSiE bhat')
p_bhat = susie_plot_locuszoom(z, mod_bhat.ld, pos = ss.dat$pos$POS/1e6, gene.pos.map = gene.pos.map, ld = R[z.max,], z.ref.name = names(z.max), title = 'UQCC1 SuSiE bhat', y.susie = 'p')
grid.arrange(pip_bhat, p_bhat, ncol=2)
```
